test/common/config/config_provider_impl_test.cc

```diff
@@ -737,9 +737,9 @@ TEST_F(DeltaConfigProviderImplTest, MultipleDeltaSubscriptions) {
   // Issue a second config update to validate that having multiple providers bound to the
   // subscription causes a single update to the underlying shared config implementation.
   subscription.onConfigUpdate(untyped_dummy_configs, "2");
-  // NOTE: the config implementation is append only and _does not_ track updates/removals to the
-  // config proto set, so the expectation is to double the size of the set.
-  EXPECT_EQ(provider1->config<const ThreadLocalDummyConfig>()->numProtos(), 4);
+  // NOTE: the two providers share the same config, each config update propagation would add 2
+  // protos to the same config's proto vector, so the expectation is 6 here.
+  EXPECT_EQ(provider1->config<const ThreadLocalDummyConfig>()->numProtos(), 6);
```

Comment:
Tests need to be modified to account for the change in `applyDeltaConfigUpdate()`'s implementation. Since the latter is no longer assuming that providers will be sharing a config, the comment in line 737 needs to be updated. Also, I suggest adding a new test that validates that providers with different configs properly reflect changes to each instance of the config (i.e., to ensure that `applyDeltaConfigUpdate()` is not memoizing the config in any way).

Reply:
It largely depends on the implementation of the ConfigProvider. 
I will make it one-config-per-provider for now as the framework doesn't enforce providers to share underlying protobuf yet. Will address the "providers share underlying proto-buf" in a following up PR.